src/v2/modules/addresses/validator.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const Joi = require('@hapi/joi');
+
+const mandatoryPostcodeCountries = [
+  'united kingdom',
+  'england',
+  'wales',
+  'scotland',
+  'northern ireland',
+  'uk'
+];
+
+// https://en.wikipedia.org/wiki/Postcodes_in_the_United_Kingdom#Validation
+const postcodeRegex = /^(([A-Z]{1,2}[0-9][A-Z0-9]?|ASCN|STHL|TDCU|BBND|[BFS]IQQ|PCRN|TKCA) ?[0-9][A-Z]{2}|BFPO ?[0-9]{1,4}|(KY[0-9]|MSR|VG|AI)[ -]?[0-9]{4}|[A-Z]{2} ?[0-9]{2}|GE ?CX|GIR ?0A{2}|SAN ?TA1)$/;
+
+const schema = Joi.object({
```

Comment:
I think we need to determine a pattern for this.  E.g. in the existing createCompany endpoint, the `type` property will fail with a 400 if you don't supply organisation|person, even if it is a valid string.
Perhaps the route level validation should only be checking for valid keys and basic types, and anything more advanced should be service level as you have it here?
In which case we will need to modify some of the other routes already completed.

Reply:
I think the service layer should impose the busiess rules because then it can be used within the CRM itself, not just from outside. So as you say, the route validation is just to check that the request is ok to continue with.